### PR TITLE
virts-623f: initial commit gist functionality

### PR DESCRIPTION
**Pull request by khyberspache**

### CODE REVIEW

File: app/contact/http.py [hunk 1]
```diff
@@ -16,6 +19,11 @@ def __init__(self, services):
         self.app.router.add_route('POST', '/instructions', self._instructions)
         self.app.router.add_route('POST', '/results', self._results)
 
+    def valid_config(self):
+        if hasattr(self.app, 'router'):
```
Comment (privateducky) [inline app/contact/http.py]: nice

File: data/contact/3dbb59fc-905c-4708-a4f2-e30bc9ac2903.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
```
Comment (privateducky) [inline data/contact/3dbb59fc-905c-4708-a4f2-e30bc9ac2903.yml]: confusing that we'd load some (GIST) c2's via YML and others (HTTP) without it..

File: hook.py [hunk 3]
```diff
@@ -1,19 +1,20 @@
...
-    data_svc = services.get('data_svc')
-    await data_svc.load_data(directory='plugins/stockpile/data')
-    await services.get('contact_svc').register(HTTP(services))
-    await services.get('data_svc').store(
+    stockpile_svc = StockpileService(services)
+    await stockpile_svc.file_svc.add_special_payload('mission.go', stockpile_svc.dynamically_compile)
+    await stockpile_svc.data_svc.load_data(directory='plugins/stockpile/data')
+    c2_configs = await stockpile_svc.load_c2_config(directory='plugins/stockpile/data/contact')
+    await stockpile_svc.contact_svc.register(HTTP(services))
+    await stockpile_svc.contact_svc.register(GIST(services, c2_configs['GIST']), active=True)
```
Comment (privateducky) [inline hook.py]: should active=True actually be the 'enabled' field from the C2 YML, instead?

File: data/contact/3dbb59fc-905c-4708-a4f2-e30bc9ac2903.yml (hunk 2 above)
Comment (khyberspache) [reply data/contact/3dbb59fc-905c-4708-a4f2-e30bc9ac2903.yml]: I agree. Updating now.

File: app/contact/gist.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        super().__init__(name=config['name'], services=services)
+        self.keys = config['config']['keys']
+        self.log = self.file_svc.create_logger('GistService')
+
+    def get_config(self):
+        """
+        Returns this C2 objects api key
+        :return: GIST api key
+        """
+        return 'c2Key', self.keys[0]
```
Comment (privateducky) [inline app/contact/gist.py]: what else calls get_config, except for valid_config?

File: hook.py (hunk 3 above)
Comment (khyberspache) [reply hook.py]: Yes, I made a stupid - updating now.

File: app/contact/gist.py (hunk 4 above)
Comment (privateducky) [reply app/contact/gist.py]: nvm I see it in the other PR

File: hook.py (hunk 3 above)
Comment (khyberspache) [reply hook.py]: active also refers to it's type - I'm going to iterate on this quickly

**Pull request merged**